### PR TITLE
Timestamp to date

### PR DESCRIPTION
**Pull request by mpneuried**
I added a script to convert unix timestamps to a date.

Open Features:
- detect and switch between seconds and miliseconds

### CODE REVIEW

File: Time/ts-convert.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export LANG="${LANG:-en_US.UTF-8}"
+
+PREFIX=""
+APPENDIX=""
+
+
+# user clicks on 'encodepaste'
+if [[ "$1" = "encodepaste" ]]; then
+  CONVERTED=$(echo -n "${PREFIX}$(pbpaste)${APPENDIX}" | xargs -I {} date -r {} +'%Y-%m-%d %H:%M:%S')
+  $(echo -n "$CONVERTED" | pbcopy)
```
Comment (keithamus) [inline Time/ts-convert.sh]: This line should not include `$()` because that attempts to execute any output.

File: Time/ts-convert.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  CONVERTED=$(echo -n "${PREFIX}$(pbpaste)${APPENDIX}" | xargs -I {} date -r {} +'%Y-%m-%d %H:%M:%S')
+  osascript -e "display notification \"$CONVERTED\" with title \"DATE:\"" &> /dev/null
+  exit
+fi
+
+# user clicks on 'now'
+if [[ "$1" = "now" ]]; then
+  NOW=$(date +%s)
+
+  $(echo -n "$NOW" | pbcopy)
```
Comment (keithamus) [inline Time/ts-convert.sh]: This line should not include `$()` because that attempts to execute any output.

**Comment by mpneuried** [comment]
i changed it to:
`CONVERTED=$(date -r `pbpaste` +'%Y-%m-%d %H:%M:%S')`

Hopefully this is a better solution ;-)

File: Time/ts-convert.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# <bitbar.image>http://i.imgur.com/dXzJNOS.jpg?1</birtbar.image>
+# <bitbar.dependencies></bitbar.dependencies>
+
+# Hack for language not being set properly and unicode support
+export LANG="${LANG:-en_US.UTF-8}"
+
+# user clicks on 'encodepaste'
+if [[ "$1" = "encodepaste" ]]; then
+  CONVERTED=$(date -r `pbpaste` +'%Y-%m-%d %H:%M:%S')
+  $(echo -n "$CONVERTED" | pbcopy)
```
Comment (keithamus) [inline Time/ts-convert.sh]: This still needs to loose the $(). So

```sh
$(echo -n "$CONVERTED" | pbcopy)
```

Needs to become
```sh
echo -n "$CONVERTED" | pbcopy
```

File: Time/ts-convert.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+if [[ "$1" = "encode" ]]; then
+  CONVERTED=$(date -r `pbpaste` +'%Y-%m-%d %H:%M:%S')
+  osascript -e "display notification \"$CONVERTED\" with title \"DATE:\"" &> /dev/null
+  exit
+fi
+
+# user clicks on 'now'
+if [[ "$1" = "now" ]]; then
+  NOW=$(date +%s)
+  $(echo -n "$NOW" | pbcopy)
```
Comment (keithamus) [inline Time/ts-convert.sh]: Same here.

**Comment by mpneuried** [comment]
@keithamus Thanks for you help.
I'm very new to bash scripting and thankful for every hint ;-)

But one question... I used [this](https://github.com/matryer/bitbar-plugins/blob/master/System/clipboard-base64-encoder.sh#L22) as inspiration.
```sh
$(echo -n "$(echo -n "${PREFIX}$(pbpaste)${APPENDIX}" | base64)" | pbcopy)
```
Because this looks quite similar i guess it has the same issues?

**Comment by keithamus** [comment]
Yeah, I probably shouldn't have merged that like it is - that will also have the same problems.

**Pull request merged**